### DISCUSSION TITLE
Github not counting contributions

### DISCUSSION THREAD

**Discussion by muhammad-Osman** [community/community#41839] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Can someone please tell me why github is not counting my contributions?

**Comment by airtower-luna** [comment]
Not without a link to the relevant commits, but you can check yourself using the documentation: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile

**Comment by samharp** [comment]
Would definitely need a link and/or more information to know for sure, but a big thing that got me was committing to a branch that was not main/master. Commits to additional branches don't count towards your contributions UNTIL you merge that branch into main/master. Hopes this helps!

**Comment by mohamedzhioua** [comment]
this usually happens due to a lot of factors, for which you must visit this [GitHub help](https://help.github.com/articles/why-are-my-contributions-not-showing-up-on-my-profile/#which-contributions-are-counted)

The most common mistakes I found were :

Email not configured properly on local machine. (your laptop/pc).
Forking.
Not using default or gh-pages branch for commit.
I personally had issue 1 recently, for which I went to my terminal/cmd/cli and configure my email address locally by running this command

git config --global user.email yourname@email.address
Once that's done, try running this config again like this:

git config --global user.email
You should be able to see:

yourname@email.address
If this was correct, you have successfully configured your local machine with your global public repo. Now your next commit will be credited towards your calendar.

**Comment by DSDmark** [comment]
Make sure that Git that includes your email address in the commit history. Contributions are tied to your email address. so if your email address is not included in the commit history, your contributions will not be credited to your profile.

Make sure that you have not configured Git to ignore certain files or directories. If you have set up Git to ignore certain files or directories, your changes to those files will not be tracked, and your contributions will not be counted.
